src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -146,14 +166,22 @@ private Completable resume(RxWebSocket ws, BotToken token) {
     Preconditions.checkState(
         lastSeenSequenceNumber.isPresent(), "Can not resume without a sequence number");
 
-    return Single.just(
-            ImmutableResume.builder()
-                .token(token)
-                .sessionId(sessionId.get())
-                .seq(lastSeenSequenceNumber.get())
-                .build())
-        .map(payloads::resume)
-        .flatMapCompletable(p -> send(ws, p));
+    return send(
```

Comment:
I'm 50/50 about this send method.

It does reduce a bit of duplication, but it also makes the method hard to read.

With the flatMapCompletable it reads in sequence: build the resume payload, apply/map the payloads resume function, and then send it. The send at the start kind of has it out of sequence a bit.

Reply:
I can rename it, i think that's the best thing to do in this situation.

Reply:
I'm thinking `completePayload`

Reply:
It's not the naming that I'm concerned about it's the flow. By having the send in the map at the end, you end up with the code reading as: start with something, do this, do that, do this, then send it. With the method you end up with: send (something, do this, do that, do this )